codegen/CMakeLists.txt

```diff
@@ -27,6 +27,22 @@ file(GLOB_RECURSE KERNEL_FILES CONFIGURE_DEPENDS
 add_embed_library(ck_headers ${KERNEL_FILES} RELATIVE ${CK_ROOT}/include)
 
 file(GLOB SOURCES CONFIGURE_DEPENDS src/*.cpp)
+
+if(INSTANCES_ONLY)
+	set(CG_TARGETS ${DEFAULT_GPU_TARGETS})
+else()
+	set(CG_TARGETS ${GPU_TARGETS})
+endif()
+
+list(REMOVE_ITEM CG_TARGETS gfx900 gfx906 gfx1030 gfx1100 gfx1101 gfx1102 gfx1103 gfx1200 gfx1201)
+set(offload_targets)
+foreach(target IN LISTS CG_TARGETS)
+	string(APPEND offload_targets "--offload-arch=${target} ")
+endforeach()
+foreach(source IN LISTS SOURCES)
+	set_source_files_properties(${source} PROPERTIES COMPILE_FLAGS ${offload_targets})
+endforeach()
```

Comment:
This should not require device compilation. This library is designed to be a host-only API. In migraphx, we will only consume it from the host.

Furthermore, it doesn't contain any gpu code, if you are wanting to add gpu code then it should go in a seperate library.

Reply:
Yes, I've verified that there's indeed no gpu code in the libraires, so we'll remove the gpu targets.